tests/helpers/k8s.bash

```diff
@@ -628,3 +629,166 @@ function kindbox_cluster_teardown() {
   # Delete cluster config.
   rm -rf /root/.kube/${cluster}-config
 }
+
+
+################################################################################
+# K3s specific functions
+################################################################################
+
+
+function k3s_all_nodes_ready() {
+  local cluster=$1
+  local num_workers=$2
+  local delay=$3
+
+  local timestamp=$(date +%s)
+  local timeout=$(( $timestamp + $delay ))
+  local all_ok
+
+  while [ $timestamp -lt $timeout ]; do
+    all_ok="true"
+
+    for (( i=1; i<=$num_workers; i++ )); do
+      master=${cluster}-master
+      worker=${cluster}-worker-${i}
+
+		run k8s_node_ready $worker
+      if [ "$status" -ne 0 ]; then
+        all_ok="false"
+        break
+      fi
+    done
+
+    if [[ "$all_ok" == "true" ]]; then
+       break
+    fi
+
+    sleep 2
+    timestamp=$(date +%s)
+  done
+
+  if [[ "$all_ok" != "true" ]]; then
+    echo 1
+  else
+    echo 0
+  fi
+}
+
+# Deploys a k3s cluster through k3sup tool. The cluster has one master node
+# and the given number of worker nodes. The cluster uses the K8s flannel cni.
+#
+# usage: k3s_cluster_setup <cluster_name> <num_workers> <network> <node_image> <k8s_version>
+#
+# cluster: name of the cluster; nodes in the cluster are named "<cluster_name>-master",
+#          "<cluster-name>-worker-1", "<cluster-name>-worker-2", etc.
+# num_workers: number of k3s worker nodes
+# network: docker network to which the k3s nodes are connected (e.g., bridge,
+#          user-defined, etc.)
+#
+function k3s_cluster_setup() {
+  local cluster=$1
+  local num_workers=$2
+  local net=$3
+  local node_image=$4
+  local k8s_version=$5
+  local cni=$6
+
+  local pod_net_cidr=10.244.0.0/16
+  local master_node=${cluster}-master
+
+  # Install k3sup.
+  curl -sLS https://get.k3sup.dev | sh
+  [ "$status" -eq 0 ]
+
+  # Create master and worker nodes.
+  run docker_run --rm --name=${master_node} --hostname=${master_node} ${node_image}
```

Comment:
I believe the `docker_run` function calls `docker` inside, which in turn calls the bats `run` macro, so `run docker_run` is redundant.

No need for the status check in the next line either.

Reply:
Done.